### PR TITLE
Migrate to from GitHub castorini/data to uWaterloo Castor-data

### PR DESCRIPTION
**Pull request by tuzhucheng**
Migrate from castorini/data to https://git.uwaterloo.ca/jimmylin/Castor-data.

* Changing paths in code to reflect new repo name.
* Remove redundant instructions in README - aim to centralize instructions like cloning in the main README.md.

@rosequ please review changes in anserini_dependency, idf_baseline, sm_cnn, and kim_cnn?
@Victor0118 please review changes in nce and mp_cnn?

Both please review changes to the main README.md?

### CODE REVIEW

File: anserini_dependency/README.md [hunk 1]
```diff
@@ -34,22 +31,13 @@ Install the dependency packages:
 
 ```
 cd Castor
-pip3 install -r requirements.txt
+pip install -r requirements.txt
```
Comment (Victor0118) [inline anserini_dependency/README.md]: Do all our models support both py2 and py3?

File: anserini_dependency/api.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ def get_answers(question, num_hits, k):
     parser.add_argument('--no_cuda', action='store_false', help='do not use cuda', dest='cuda')
     parser.add_argument('--gpu', type=int, default=0) # Use -1 for CPU
     parser.add_argument('--seed', type=int, default=3435)
-    parser.add_argument('--dataset', help="the QA dataset folder {TrecQA|WikiQA}", default='../../data/TrecQA/')
+    parser.add_argument('--dataset', help="the QA dataset folder {TrecQA|WikiQA}", default='../../Castor-data/TrecQA/')
```
Comment (Victor0118) [inline anserini_dependency/api.py]: two argument parsers in a file?

File: sm_cnn/README.md [hunk 3]
```diff
@@ -144,7 +96,7 @@ NB: The results on WikiQA are based on the SM model hyperparameters.
 to the `data/` folder
 
 ```bash
-python $PYTHONPATH/utils/build_w2v.py --input data/aquaint+wiki.txt.gz.ndim=50.bin
+python $PYTHONPATH/utils/build_w2v.py --input ../../Castor-data/word2vec/embeddings/aquaint+wiki.txt.gz.ndim=50.bin
```
Comment (Victor0118) [inline sm_cnn/README.md]: different from the path before `embeddings/word2vec`?

File: nce/nce_pairwise_sm/args.py [hunk 4]
```diff
@@ -20,7 +20,7 @@ def get_args():
     parser.add_argument('--words_dim', type=int, default=50)
     parser.add_argument('--dropout', type=float, default=0.5)
     parser.add_argument('--epoch_decay', type=int, default=15)
-    parser.add_argument('--wordvec_dir', type=str, default='../../../data/word2vec/')
+    parser.add_argument('--wordvec_dir', type=str, default='../../../Castor-data/embeddings/word2vec/')
```
Comment (Victor0118) [inline nce/nce_pairwise_sm/args.py]: different from the path below `word2vec/embeddings` ?

File: anserini_dependency/README.md (hunk 1 above)
Comment (tuzhucheng) [reply anserini_dependency/README.md]: No, it only supports Python 3. But if they followed the setup correctly `pip` should be the correct one by default - if they explicitly use `pip3` and `pip` doesn't work something is wrong with their environment set-up.

File: sm_cnn/README.md (hunk 3 above)
Comment (tuzhucheng) [reply sm_cnn/README.md]: Yeah, this is the correct behaviour. We moved `GloVe`, `word2vec` etc. to `embeddings` per @lintool's suggestion.

File: nce/nce_pairwise_sm/args.py (hunk 4 above)
Comment (tuzhucheng) [reply nce/nce_pairwise_sm/args.py]: Ditto, embeddings are moved to the `embeddings` subdirectory now.

File: anserini_dependency/api.py (hunk 2 above)
Comment (tuzhucheng) [reply anserini_dependency/api.py]: Didn't understand this comment :(

Comment (tuzhucheng) [reply anserini_dependency/api.py]: I see - that parser was already there (unchanged except for default path arg). This PR is only about the Castor-data issue. If that is a bug I would prefer to fix it separately in another PR.

**Pull request merged**